modules/install/k3s_master.sh

```diff
@@ -93,22 +91,43 @@ disable_cloud_setup() {
    fi
 }
 
-install() {
+export_variables() {
   export "$install_mode"="$version"
+   alias k=kubectl
+}
+
+install_k3s() {
+  url="https://get.k3s.io"
+  params="INSTALL_K3S_TYPE='server'"
 
-  if [ "$datastore_type" = "etcd" ]; then
     if [[ -n "$channel" ]]; then
-      curl -sfL https://get.k3s.io | INSTALL_K3S_CHANNEL=$channel INSTALL_K3S_TYPE='server' sh -s - server
-    else
-      curl -sfL https://get.k3s.io | INSTALL_K3S_TYPE='server' sh -s - server
+       params="$params INSTALL_K3S_CHANNEL=$channel"
     fi
-  elif  [[ "$datastore_type" = "external" ]]; then
-    if [[ -n "$channel" ]]; then
-      curl -sfL https://get.k3s.io | INSTALL_K3S_CHANNEL=$channel sh -s - server --datastore-endpoint="$datastore_endpoint"
+
+    if [ "$datastore_type" = "etcd" ]; then
+        install_command="curl -sfL $url | $params sh -s - server"
+    elif [[ "$datastore_type" = "external" ]]; then
+        install_command="curl -sfL $url | $params sh -s - server --datastore-endpoint=\"$datastore_endpoint\""
+    fi
+
+  eval "$install_command"
+}
+
+check_service() {
+    if systemctl is-active --quiet k3s; then
+         echo "K3s is running on server node ip $public_ip"
     else
-      curl -sfL https://get.k3s.io | sh -s - server  --datastore-endpoint="$datastore_endpoint"
+         printf  "K3s failed to start on server node ip %s\n" "$public_ip"
+         sudo journalctl -xeu k3s.service | grep -i "failed\|fatal"
+         exit 1
     fi
-  fi
+}
+
+install() {
+  export_variables
+  install_k3s
+  sleep 20
```

Comment:
We need to handle this differently if wait is needed. Adds to install time which may not be necessary

Reply:
i decrease to 10 s, but this is just a time to wait the install complete. nothing more to do here